pyccel/ast/builtin_methods/list_methods.py

```diff
@@ -306,3 +307,41 @@ def __init__(self, list_obj) -> None:
         self._order = list_obj.order
         self._class_type = list_obj.class_type
         super().__init__(list_obj)
+
+#==============================================================================
+class ListSort(ListMethod) :
+    """
+    Represents a call to the .sort() method.
+
+    Represents a call to the `.sort()` method, which sorts the elements of the
+    list in ascending order and modifies the original list in place. This means
+    that the elements of the original list are rearranged to be in sorted order.
+    Note that the .sort() method doesn't return any value.
+    
+    >>> a = [3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5]
+    >>> a.sort()
+    >>> print(a)
+    [1, 1, 2, 3, 3, 4, 5, 5, 5, 6, 9]
+
+    Parameters
+    ----------
+    list_obj : TypedAstNode
+        The list object which the method is called from.
+
+    reverse : TypedAstNode
+        Optional, reverse=True will sort the list descending. Default is reverse=False.
+
+    key : TypedAstNode
+        Optional, a function to specify the sorting criteria(s).
+    """
+    __slots__ = ()
+    _rank = 0
+    _order = None
+    _shape = None
+    _class_type = VoidType()
+    name = 'sort'
+
+    def __init__(self, list_obj, reverse=None, key=None) -> None:
+        if reverse is not None or key is not None:
+            raise TypeError("Optional Parameters are not supported for sort() method.")
```

Comment:
Please mention this restriction somewhere in the docstring

Reply:
updated 35a6df89c347a11eba1dbd61acd0317e55666844